Duplicate identifier 'MochaDone' when using @types/mocha

I have mocha-typescript@latest and types/mocha@latest as dev dependencies in my package.json.

When I compile my *.ts file containing the tests, I get the following error messages from the typescript compiler:

```
node_modules/@types/mocha/index.d.ts(56,6): error TS2300: Duplicate identifier 'MochaDone'.
node_modules/mocha-typescript/index.d.ts(15,11): error TS2300: Duplicate identifier 'MochaDone'.
```

Any idea how to fix this is appreciated.

@hubbergit good find, here is even more:

```
node_modules/@types/mocha/index.d.ts: error TS2451: Cannot redeclare block-scoped variable 'describe'.
node_modules/@types/mocha/index.d.ts: error TS2451: Cannot redeclare block-scoped variable 'suite'.
node_modules/@types/mocha/index.d.ts: error TS2451: Cannot redeclare block-scoped variable 'it'.
node_modules/@types/mocha/index.d.ts: error TS2451: Cannot redeclare block-scoped variable 'test'.
node_modules/@types/mocha/index.d.ts: error TS2300: Duplicate identifier 'MochaDone'.
node_modules/mocha-typescript/globals.d.ts: error TS2451: Cannot redeclare block-scoped variable 'suite'.
node_modules/mocha-typescript/globals.d.ts: error TS2451: Cannot redeclare block-scoped variable 'test'.
node_modules/mocha-typescript/globals.d.ts: error TS2451: Cannot redeclare block-scoped variable 'describe'.
node_modules/mocha-typescript/globals.d.ts: error TS2451: Cannot redeclare block-scoped variable 'it'.
node_modules/mocha-typescript/index.d.ts: error TS2300: Duplicate identifier 'MochaDone'.
```

--- a/globals.d.ts
+++ b/globals.d.ts
@@ -48,12 +48,6 @@ declare namespace Mocha {
     }
 }
 
-declare var suite: Mocha.IContextDefinition;
-declare var test: Mocha.ITestDefinition;
-
-declare var describe: Mocha.IContextDefinition;
-declare var it: Mocha.ITestDefinition;
-
 declare var skipOnError: MochaTypeScript.SuiteTrait;
 
 declare function slow(time: number): PropertyDecorator & ClassDecorator & MochaTypeScript.SuiteTrait & MochaTypeScript.TestTrait;

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,19 +1,10 @@
 declare namespace Mocha {
-    export interface IContextDefinition {}
-    export interface ITestDefinition {}
-    export interface ISuiteCallbackContext {}
-    export interface ITestCallbackContext {}
-
     export interface ITest {
         <T>(target: Object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>): TypedPropertyDescriptor<T> | void;
     }
     export interface ISuite {
         <TFunction extends Function>(target: TFunction): TFunction | void;
     }
-}
-
-interface MochaDone {
-    (error?: any): any;
 }
 
 declare namespace MochaTypeScript {

--- a/index.ts
+++ b/index.ts
@@ -74,12 +74,12 @@ function applyDecorators(mocha: Mocha.IHookCallbackContext | Mocha.ISuiteCallbac
 		mocha.timeout(timeoutValue);
 	}
 	const slowValue = method[slowSymbol];
-	if (typeof slowValue === "number") {
-		mocha.slow(slowValue);
+	if (mocha['slow'] && typeof slowValue === "number") {
+		mocha['slow'](slowValue);
 	}
 	const retriesValue = method[retriesSymbol];
-	if (typeof retriesValue === "number") {
-		mocha.retries(retriesValue);
+	if (mocha['retries'] && typeof retriesValue === "number") {
+		mocha['retries'](retriesValue);
 	}
 	const contextProperty = ctorOrProto[contextSymbol];
 	if (contextProperty) {

--- a/package.json
+++ b/package.json
@@ -25,6 +25,7 @@
   },
   "devDependencies": {
     "@types/chai": "^3.4.35",
+    "@types/mocha": "^5.0.0",
     "@types/node": "^7.0.12",
     "assert": "^1.3.0",
     "better-assert": "^1.0.2",

